src/Controls/src/Core/Platform/Android/Extensions/FormattedStringExtensions.cs

```diff
@@ -133,31 +133,28 @@ public static SpannableString ToSpannableString(this Label label)
 
 		public static void RecalculateSpanPositions(this TextView textView, Label element, SpannableString spannableString, SizeRequest finalSize)
 		{
-			if (element?.FormattedText?.Spans == null || element.FormattedText.Spans.Count == 0)
+			if (element?.FormattedText?.Spans is null || element.FormattedText.Spans.Count == 0)
 				return;
 
 			var labelWidth = finalSize.Request.Width;
-
 			if (labelWidth <= 0 || finalSize.Request.Height <= 0)
 				return;
 
 			if (spannableString == null || spannableString.IsDisposed())
 				return;
 
 			var layout = textView.Layout;
-
 			if (layout == null)
 				return;
 
 			int next = 0;
 			int count = 0;
-			IList<int> totalLineHeights = new List<int>();
 
-#pragma warning disable CA1416
-			var strlen = spannableString.Length();
-#pragma warning restore CA1416
+			var padding = element.Padding;
+			var padLeft = (int)textView.Context.ToPixels(padding.Left);
+			var padTop = (int)textView.Context.ToPixels(padding.Top);
 
-			for (int i = 0; i < strlen; i = next)
+			for (int i = 0; i < spannableString.Length(); i = next)
```

Comment:
I think the length is extracted as it is quite expensive to get as it has to cross the JNI bridge.

Reply:
Yeah, good point. Reverted this change.